unittests/testassetattachments.py

```diff
@@ -0,0 +1,214 @@
+"""
+Test archivist
+"""
+
+from io import BytesIO
+from unittest import TestCase, mock
+
+from archivist.archivist import Archivist
+from archivist.constants import (
+    ROOT,
+    ASSETS_LABEL,
+    ATTACHMENTS_LABEL,
+    ASSETATTACHMENTS_LABEL,
+    ASSETATTACHMENTS_SUBPATH,
+)
+
+from .mock_response import MockResponse
+
+# pylint: disable=protected-access
+
+
+PROPS = {
+    "hash": {"alg": "SHA256", "value": "xxxxxxxxxxxxxxxxxxxxxxx"},
+    "mime_type": "image/jpeg",
+    "timestamp_accepted": "2019-11-07T15:31:49Z",
+    "size": 31424,
+}
+ASSET_UUID = "b2678528-0136-4876-ad56-904e12c4b4c6"
+ASSET_ID = f"{ASSETS_LABEL}/{ASSET_UUID}"
+ATTACHMENT_UUID = "abcdef28-0136-4876-ad56-904e12c4b4c6"
+ATTACHMENT_ID = f"{ATTACHMENTS_LABEL}/{ATTACHMENT_UUID}"
+SUBPATH = (
+    f"{ASSETATTACHMENTS_SUBPATH}/{ASSETATTACHMENTS_LABEL}/{ASSET_ID}/{ATTACHMENT_UUID}"
+)
+RESPONSE = {
+    **PROPS,
+    "identity": ATTACHMENT_ID,
+}
+INFO = {
+    "hash": {
+        "alg": "SHA256",
+        "value": "e1105070ba828007508566e28a2b8d4c65d192e9eaf3b7868382b7cae747b397",
+    },
+    "identity": ATTACHMENT_ID,
+    "issuer": "",
+    "mime_type": "application/zip",
+    "size": 308,
+    "subject": "",
+    "tenantid": "",
+    "timestamp_accepted": "2022-03-01T09:25:43Z",
+    "scanned_status": "NOT_SCANNED",
+    "scanned_bad_reason": "",
+    "scanned_timestamp": "",
+}
+
+
+class TestAssetAttachmentsBase(TestCase):
+    """
+    Test Archivist Attachments Create method
+    """
+
+    maxDiff = None
+
+    def setUp(self):
+        self.arch = Archivist("url", "authauthauth")
+
+    def test_assetattachments_str(self):
+        """
+        Test attachments str
+        """
+        self.assertEqual(
+            str(self.arch.assetattachments),
+            "AssetAttachmentsClient(url)",
+            msg="Incorrect str",
+        )
+
+
+class TestAssetAttachmentsDownload(TestAssetAttachmentsBase):
+    """
+    Test Archivist Attachments Downloads method
+    """
+
+    maxDiff = None
+
+    def common_assetattachments_download(
+        self,
+        input_params,
+        expected_params,
```

Comment:
Under what circumstances would these be different? All callers seem to pass a redundant pair in, and in general it seems like expected would always be the same as input unless you're explicitly looking for a weird failure case.

Reply:
in one case None is passed in but an empty dict is returned.